controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -482,7 +481,7 @@ public Application deleteApplication(ApplicationId id, Optional<NToken> token) {
         try (Lock lock = lock(id)) {
             Optional<Application> application = get(id);
             if ( ! application.isPresent()) return null;
-            if ( ! application.get().deployments().isEmpty())
+            if ( ! application.get().productionDeployments().isEmpty())
```

Comment:
Won't this allow application data in the controller to be deleted while there are still active dev or perf deployments? And potentially make config servers and controller out of sync?

Reply:
It is ok to delete them, but I see the deployments are not deleted here. Nice catch, Incoming PR then.

Reply:
Deleting and deactivating a deployment is separate code path / API call I think.

Reply:
Yes. For non-prod a recursive delete would be nice, but probably not worth the bother.